addons/project/controllers/portal.py

```diff
@@ -94,40 +93,102 @@ def portal_my_project(self, project_id=None, access_token=None, **kw):
+            project_accessible = bool(task.project_id.id and self._document_check_access('project.project', task.project_id.id))
+        except (AccessError, MissingError):
+            project_accessible = False
+
         values = {
             'page_name': 'task',
             'task': task,
-            'user': request.env.user
+            'user': request.env.user,
+            'project_accessible': project_accessible
         }
         return self._get_page_view_values(task, access_token, values, 'my_tasks_history', False, **kwargs)
 
-    @http.route(['/my/tasks', '/my/tasks/page/<int:page>'], type='http', auth="user", website=True)
-    def portal_my_tasks(self, page=1, date_begin=None, date_end=None, sortby=None, filterby=None, search=None, search_in='content', groupby=None, **kw):
-        values = self._prepare_portal_layout_values()
-        searchbar_sortings = {
+    def _task_get_searchbar_sortings(self):
+        return {
             'date': {'label': _('Newest'), 'order': 'create_date desc'},
             'name': {'label': _('Title'), 'order': 'name'},
             'stage': {'label': _('Stage'), 'order': 'stage_id, project_id'},
             'project': {'label': _('Project'), 'order': 'project_id, stage_id'},
             'update': {'label': _('Last Stage Update'), 'order': 'date_last_stage_update desc'},
+            'date_deadline': {'label': _('Deadline'), 'order': 'date_deadline asc'},
         }
-        searchbar_filters = {
-            'all': {'label': _('All'), 'domain': []},
+
+    def _task_get_searchbar_groupby(self):
+        values = {
+            'none': {'input': 'none', 'label': _('None'), 'order': 1},
+            'project': {'input': 'project', 'label': _('Project'), 'order': 2},
+            'stage': {'input': 'stage', 'label': _('Stage'), 'order': 3},
+            'user': {'input': 'user', 'label': _('Assigned to'), 'order': 6},
+            'customer': {'input': 'customer', 'label': _('Customer'), 'order': 7}
         }
-        searchbar_inputs = {
-            'content': {'input': 'content', 'label': Markup(_('Search <span class="nolabel"> (in Content)</span>'))},
-            'message': {'input': 'message', 'label': _('Search in Messages')},
-            'customer': {'input': 'customer', 'label': _('Search in Customer')},
-            'stage': {'input': 'stage', 'label': _('Search in Stages')},
-            'project': {'input': 'project', 'label': _('Search in Project')},
-            'all': {'input': 'all', 'label': _('Search in All')},
+        return dict(sorted(values.items(), key=lambda item: item[1]["order"]))
+
+    def _task_get_groupby_mapping(self):
+        return {
+            'project': 'project_id',
+            'stage': 'stage_id',
+            'user': 'user_id',
+            'customer': 'partner_id'
+        }
+
+    def _task_get_order(self, order, groupby):
+        groupby_mapping = self._task_get_groupby_mapping()
+        return '%s, %s' % (groupby_mapping[groupby], order)
```

Comment:
If we do not have any groupby, then we should return only the order value. That is:
```suggestion
        groupby_mapping = self._task_get_groupby_mapping()
        field_name = groupby_mapping.get(groupby, '')
        if not field_name:
            return order
        return '%s, %s' % (field_name, order)
```
Sorry, I have not tested before sending the suggestion. Normally, it could be solved your error displayed in the runbot. 🙂